python/test/unit/test_address_sanitizer.py

```diff
@@ -0,0 +1,18 @@
+import os
+import subprocess
+import pytest
+
+from triton._internal_testing import is_hip
+
+os.environ["AMD_PYTORCH_NO_CUDA_MEMORY_CACHING"] = "1"
```

Comment:
Can you add comments to explain why some of the env vars are needed if not obvious? It's a good documentation we can point folks to later.